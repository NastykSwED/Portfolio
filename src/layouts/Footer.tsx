export const Footer = () => {
	return (
		<footer className='re-Footer' id='re-Footer'>
			<div className='re-Footer-head'>
				<div className='re-Footer-content'>
					<h1 className='re-Footer-title'>Ronald Torres</h1>
					<h2 className='re-Footer-subTitle'>Front End Developer</h2>
				</div>
				<div className='re-Footer-enjoyer'>
					<span>React Enjoyer</span>
				</div>
			</div>
			<div className='re-Footer-legal'>
				<div className='re-Footer-disclaimer'>
					<p className='re-Footer-disclaimerTitle'>
						Â©2022 Ronald Torres. All rights reserved
					</p>
				</div>
			</div>
		</footer>
	);
};
