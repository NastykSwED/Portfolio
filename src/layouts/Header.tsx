import { useState } from 'react';

import { Navbar, Hamburger } from '../components';

export const Header = () => {
	const [isActive, setIsActive] = useState<boolean>(false);

	const onClick = () => {
		setIsActive(!isActive);
	};

	return (
		<header className='re-Header' id='re-Header'>
			<div className='re-Header-wrapNavbar'>
				<h1 className='re-Header-logo'>Ronald Torres</h1>
				<Navbar isActive={isActive} />
				<Hamburger onClick={onClick} />
			</div>
		</header>
	);
};
