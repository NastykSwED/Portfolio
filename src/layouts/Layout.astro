---
import '../sass/styles.scss';
import '@fontsource/raleway';
import { SEO } from 'astro-seo';
---

<!DOCTYPE html>
<html lang='en' data-theme='light'>
	<head>
		<!-- Astro Lover -->
		<SEO
			charset='UTF-8'
			title='Ronald Torres | Front End Developer'
			description="Hi I'm Ronald Torres. A Front End Developer and CS Student with experience in web development that enjoys web technologies"
			canonical='https://nastykswed.dev/'
			openGraph={{
				basic: {
					url: 'https://nastykswed.dev/',
					type: 'website',
					title: 'Ronald Torres',
					image: '/preview.png',
				},
				optional: {
					siteName: 'nastykswed.dev',
					description:
						"Hi I'm Ronald Torres. A Front End Developer and CS Student with experience in web development that enjoys web technologies",
				},
				image: {
					alt: 'Ronald Torres Portfolio Preview',
				},
			}}
			extend={{
				link: [
					{ rel: 'icon', type: 'image/x-icon', href: '/favicon.ico' },
					{ rel: 'manifest', href: '/manifest.json' },
				],
				meta: [
					{
						content: 'width=device-width, initial-scale=1.0',
						name: 'viewport',
					},
					{
						name: 'twitter:card',
						content: 'summary_large_image',
					},
					{
						property: 'twitter:domain',
						content: 'nastykswed.dev',
					},
					{
						property: 'twitter:url',
						content: 'https://nastykswed.dev/',
					},
					{
						name: 'twitter:title',
						content: 'Ronald Torres',
					},
					{
						name: 'twitter:description',
						content:
							"Hi I'm Ronald Torres. A Front End Developer and CS Student with experience in web development that enjoys web technologies",
					},
					{
						name: 'twitter:image',
						content: '/preview.png',
					},
					{
						name: 'application-name',
						content: 'nastykswed.dev',
					},
					{
						name: 'apple-mobile-web-app-title',
						content: 'nastykswed.dev',
					},
					{
						name: 'theme-color',
						content: '#ffffff',
					},
				],
			}}
		/>
		<meta http-equiv='X-UA-Compatible' content='IE=edge' />
		<link rel='apple-touch-icon' sizes='180x180' href='/apple-touch-icon.png' />
		<link rel='icon' type='image/png' sizes='32x32' href='/favicon-32x32.png' />
		<link rel='icon' type='image/png' sizes='16x16' href='/favicon-16x16.png' />
	</head>
	<body>
		<slot />
		<script>
			const theme = localStorage.getItem('data-theme');

			if (theme == 'dark') {
				document.documentElement.setAttribute('data-theme', theme);

				const iconTheme = document.querySelector(
					'.IconTheme'
				) as HTMLImageElement;

				iconTheme.src = 'assets/icons/sun.svg';

				iconTheme.classList.remove('Icon--moon');
				iconTheme.classList.add('Icon--sun');
				iconTheme.classList.add('isActive');
			}
		</script>
	</body>
</html>
