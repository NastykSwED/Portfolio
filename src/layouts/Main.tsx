import {
	ContactSection,
	HomeSection,
	ProjectsSection,
	SkillsSection,
} from '../components';

export const Main = () => {
	return (
		<main className='re-Main' id='re-Main'>
			<HomeSection />
			<SkillsSection />
			<ProjectsSection />
			<ContactSection />
		</main>
	);
};
