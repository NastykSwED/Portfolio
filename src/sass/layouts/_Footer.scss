@use '../utils' as *;

.re-Footer {
	padding: 1rem 2rem;
	text-align: center;
	background-color: var(--bg-clr);
	box-shadow: var(--box-shadow);

	@include mediaQuery(640px) {
		padding: 1rem;
	}

	&-head {
		max-width: var(--mx-w-footer);
		margin: 0 auto;
		display: flex;
		justify-content: space-between;
		align-items: center;
		color: var(--title-clr);

		@include mediaQuery(640px) {
			flex-direction: column;
			margin-bottom: 1rem;
		}
	}

	&-content {
		color: var(--logo-clr);
		transition: 0.3s all ease-in-out;

		@include mediaQuery(640px) {
			margin-bottom: 1rem;
		}
	}

	&-title {
		margin: 0;

		@include mediaQuery(640px) {
			font-size: 2rem;
		}
	}

	&-subTitle {
		margin: 0;
		font-size: 1rem;
		font-weight: var(--font-medium);
	}

	&-enjoyer {
		font-weight: var(--font-medium);
		color: var(--logo-clr);
		transition: 0.3s all ease-in-out;
	}

	&-legal {
		color: var(--title-clr);
		transition: 0.3s all ease-in-out;
	}

	&-disclaimer {
		color: var(--logo-clr);
		transition: 0.3s all ease-in-out;
		&Title {
			margin: 0;
			font-size: 0.8rem;
			font-weight: var(--font-semi-bold);
		}
	}
}
