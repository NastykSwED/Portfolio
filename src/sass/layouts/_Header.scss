@use '../utils' as *;

.re- {
	&Header {
		position: fixed;
		z-index: var(--z-fixed);
		top: 0;
		left: 0;
		right: 0;
		background-color: var(--bg-clr);
		box-shadow: var(--box-shadow);

		&-wrapNavbar {
			display: flex;
			max-width: var(--mx-w-wrapNav);
			margin: auto;
			align-items: center;
			justify-content: space-between;
			padding: 1rem 2rem;
		}

		&-logo {
			color: var(--logo-clr);
			user-select: none;
      transition: 0.3s all ease-in-out;

			@include mediaQuery(640px) {
				font-size: 1.7rem;
			}
		}
	}
}
