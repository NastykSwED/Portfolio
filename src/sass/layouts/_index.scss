@forward './Header';
@forward './Footer';
@forward './Main';
