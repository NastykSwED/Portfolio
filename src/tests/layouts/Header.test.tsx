import { describe, it, expect } from 'vitest';

import { screen, render, fireEvent } from '@testing-library/react';

import { Header } from '../../layouts';

describe('Testing Header Layout', () => {
	it('Should match the snapshot', () => {
		const { container } = render(<Header />);

		expect(container).toMatchSnapshot();
	});

	it('Should activate the responsive navigation bar.', () => {
		render(<Header />);

		const hamburger = screen.getByTestId('Hamburger') as HTMLElement;
		const navElement = screen.getByRole('navigation') as HTMLElement;

		fireEvent.click(hamburger);

		expect(navElement.className.includes('isActive')).toBeTruthy();
	});
});
