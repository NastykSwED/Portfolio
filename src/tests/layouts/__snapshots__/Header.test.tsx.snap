// Vitest Snapshot v1

exports[`Testing Header Layout > Should match the snapshot 1`] = `
<div>
  <header
    class="re-Header"
    id="re-Header"
  >
    <div
      class="re-Header-wrapNavbar"
    >
      <h1
        class="re-Header-logo"
      >
        Ronald Torres
      </h1>
      <nav
        class="re-Navbar "
      >
        <ul
          class="re-Navbar-list"
        >
          <li
            class="re-Navbar-listItem"
          >
            <a
              class="re-Navbar-link"
              href="#re-HomeSection"
            >
              Home
            </a>
          </li>
          <li
            class="re-Navbar-listItem"
          >
            <a
              class="re-Navbar-link"
              href="#re-SkillsSection"
            >
              Skills
            </a>
          </li>
          <li
            class="re-Navbar-listItem"
          >
            <a
              class="re-Navbar-link"
              href="#re-ProjectsSection"
            >
              Projects
            </a>
          </li>
          <li
            class="re-Navbar-listItem"
          >
            <a
              class="re-Navbar-link"
              href="#re-ContactSection"
            >
              Contact me
            </a>
          </li>
        </ul>
        <div
          class="re-Navbar-CV"
        >
          <button
            class="Button Button--primary"
          >
            CV
            <span
              class="Button-Icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                role="img"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title />
                <path
                  d="M2.666 0v24h18.668V8.666l-2.668 2.668v10H5.334V2.668H10L12.666 0zm10.668 0v8h8V5.334h-2.668V2.668H16V0"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="re-Navbar-icon"
        >
          <img
            alt="Theme Icon"
            class="Icon Icon--sun isActive"
            src="/assets/icons/sun.svg"
          />
        </div>
      </nav>
      <div
        class="re-Hamburger "
        data-testid="Hamburger"
      >
        <span
          class="re-Hamburger-bar"
        />
        <span
          class="re-Hamburger-bar"
        />
        <span
          class="re-Hamburger-bar"
        />
      </div>
    </div>
  </header>
</div>
`;
